### PR TITLE
add task solution/catalog

### PR DESCRIPTION
**Pull request by vitvadym**
[DEMO LINK](https://vitvadym.github.io/layout_catalog/)
[TEST REPORT LINK](https://vitvadym.github.io/layout_catalog/report/html_report/)

### CODE REVIEW

File: src/blocks/card.scss [hunk 1]
```diff
@@ -0,0 +1,81 @@
+@import "../utils/function";
```
Comment (silchyn) [inline src/blocks/card.scss:1]: shouldn't be there

File: src/blocks/header.scss [hunk 2]
```diff
@@ -0,0 +1,56 @@
+@import "../utils/mixins";
+
+.logo-img {
```
Comment (silchyn) [inline src/blocks/header.scss:3]: this is not part of `header` block

File: src/blocks/header.scss [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+.header {
+  display: flex;
+  @include set-flex-property;
+  padding: 0 50px;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.05);
+}
+
+.nav {
```
Comment (silchyn) [inline src/blocks/header.scss:14]: same

File: src/utils/_reset.scss [hunk 4]
```diff
@@ -0,0 +1,32 @@
+* {
+  box-sizing: border-box;
+}
+
+html {
+  font-family: "Roboto", sans-serif;
+  font-style: normal;
+}
```
Comment (silchyn) [inline src/utils/_reset.scss:8]: that is not really resetting css

**Review by silchyn — CHANGES_REQUESTED** [review]

File: src/utils/_mixins.scss [hunk 5]
```diff
@@ -0,0 +1,21 @@
+@mixin set-flex-property {
+  justify-content: space-between;
+  align-items: center;
+}
```
Comment (naziamoff) [inline src/utils/_mixins.scss:4]: use more descriptive names for your mixins. I should understand exactly what it does without the need to look for its implementation. But I would add display: flex here as well

File: src/blocks/header.scss [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
```
Comment (naziamoff) [inline src/blocks/header.scss:12]: yup, this is not a header block

File: src/utils/_mixins.scss [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+@mixin buy-btn {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  font-weight: 700;
+  font-size: 14px;
+  height: 40px;
+  line-height: 16px;
+  border-radius: 5px;
+}
```
Comment (naziamoff) [inline src/utils/_mixins.scss:15]: do you really need this mixin? By its name and current usage I don't think this is reusable one

image

File: src/utils/_mixins.scss [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+@mixin container-property {
+  display: flex;
+  flex-wrap: wrap;
+  margin: 0 auto;
+}
```
Comment (naziamoff) [inline src/utils/_mixins.scss:21]: same here, totally not clear what is it about and whether you really need it. Does not seem to be reusable

File: src/index.html [hunk 9]
```diff
@@ -4,9 +4,465 @@
...
+              video
+            </a>
+          </li>
+        </ul>
+      </nav>
+    </header>
+
+    <main class="main">
+      <section class="catalog main__catalog">
+
```
Comment (naziamoff) [inline src/index.html:80]: no spaces between elements of different levels of nesting

File: src/index.html [hunk 10]
```diff
@@ -4,9 +4,465 @@
...
+          <a
+            href="#"
+            class="card__buy card__buy--hover"
+            data-qa="card-hover"
+          >
+            Buy
+          </a>
+        </article>
+
```
Comment (naziamoff) [inline src/index.html:464]: same here, those can't have empty line in between

**Review by naziamoff — CHANGES_REQUESTED** [review]
Not bad, but let's improve it a little

File: src/blocks/logo.scss [hunk 11]
```diff
@@ -0,0 +1,3 @@
+.logo-img {
```
Comment (silchyn) [inline src/blocks/logo.scss:1]: name of the file is not `logo-img`

File: src/blocks/star.scss [hunk 12]
```diff
@@ -0,0 +1,9 @@
+.star {
```
Comment (silchyn) [inline src/blocks/star.scss:1]: isn't it element of the block `stars`?

**Review by silchyn — CHANGES_REQUESTED** [review]

File: src/blocks/stars.scss [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+  display: flex;
+
+  @for $starCount from 1 to 5 {
+    &--#{$starCount} .star:nth-child(-n + #{$starCount}) {
+      background-image: url("../images/star-active.svg");
+    }
+  }
+}
+
+.star {
```
Comment (silchyn) [inline src/blocks/stars.scss:11]: this is currently not part of the `stars` block

**Review by silchyn — CHANGES_REQUESTED** [review]
Do not forget to fix all comments

File: src/blocks/stars.scss (hunk 13 above)
Comment (vitvadym) [reply src/blocks/stars.scss:11]: image
i'm totally confused

**Review by naziamoff — APPROVED** [review]

**Pull request closed without merging**